clearwater-config-manager.root/usr/share/clearwater/clearwater-config-manager/scripts/backup_config

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Project Clearwater - IMS in the Cloud
+# Copyright (C) 2015 Metaswitch Networks Ltd
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+if [ -z $1 ]
+then
+  echo "You must provide a location to backup the configuration to"
+  echo
+  echo "Usage: $0 backup_directory"
+  exit 1
+fi
+
+local_site_name=site1
+etcd_key=clearwater
+. /etc/clearwater/local_config
+
+backup_directory=$1
+
+top_key=/${etcd_key}/${local_site_name}/configuration/
+
+while read -r key; do
+  if [ "$key" != "apply_config" ]; then
```

Comment:
Current best suggestion on how to do this:

Create a /usr/share/clearwater/clearwater-config-manager/backup_markers directory

Have each plugin create an empty file in here, with the name being the key which needs backing up and restoring.

Alternatively, we could have the plugin python itself expose it, and then make this script a Python script to access the data.